src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -2066,6 +2060,10 @@ private CallResult ExecuteCall<TTracingInstructions>(EvmState vmState, byte[]? p
                     }
             }
 
+            if (gasAvailable < 0)
```

Comment:
What about underflow?

Reply:
Don't think any operation can charge 1.8 * 10^19 gas?

Reply:
Max would be SHA3 at 1.3 × 10^10 (2^31*6+30) for a 2 GiB SHA3 op

Other methods like create; sstore etc have their own checks

Reply:
Also, it could potentially change the workflow and traces? You are manipulating the stack now.

Reply:
Out of gas doesn't report stack; only changes I can see is it might report `InvalidJumpDestination` or `AccessViolation` as the error instead of out of gas for a couple instructions as it now checks those conditions first; though would fail at the same instruction

Reply:
That could be seen as more helpful as if you increased the gas, its still going to fail 🤷‍♂️